src/grid.jl

```diff
@@ -0,0 +1,90 @@
+# Grid types
+export Node, Cell, Grid
+
+# Cell typealias
+export Line, QuadraticLine,
+       Triangle, QuadraticTriangle, Quadrilateral, QuadraticQuadrilateral,
+       Tetrahedron, QuadraticTetrahedron, Hexahedron, QuadraticHexahedron
+
+# Grid utilities
+export cells, nodes, cell_coordinates, n_cells, n_nodes
+
+#########################
+# Main types for meshes #
+#########################
+"""
+A `Node` is a point in space.
+"""
+immutable Node{dim, T}
+    x::Vec{dim, T}
+end
+
+"""
+A `Cell` is a sub-domain defined by a collection of `Node`s as vertices.
+"""
+immutable Cell{dim, N}
+    id::Int
+    nodes::NTuple{N, Int}
+end
+
+"""
+A `Grid` is a collection of `Cells` and `Node`s which covers the computational domain.
+"""
+immutable Grid{dim, N, T <: Real}
+    cells::Vector{Cell{dim, N}}
+    nodes::Vector{Node{dim, T}}
+    # cell_sets # Dict? with symbol/name => Vector of id's
```

Comment:
`Dict{String, Vector{Int}}` for cell sets and node sets seems good. Perhaps worth to have a boundary sets like `Dict{String, Vector{Tuple{Int,Int}}` where first int is cell and second is edge number.